### PR TITLE
Improve offset correction.

### PR DESCRIPTION
**Pull request by ablot**
See https://github.com/SainsburyWellcomeCentre/StitchIt/issues/145

Feel free to change dramatically or reject brutally.

One of the thing I wasn't sure how to do was the offset "caching". I did not find the equivalent of a tileStats for the whole acquisition so I just created tiny mat files with one number.

### CODE REVIEW

**Comment by raacampbell** [comment]
Thanks, I will look.

**Comment by raacampbell** [comment]
Looks good. I think what I will do is:
1. Add an INI file option to choose what will be the offset. 
2. Save the offset value(s) in the average image files.

**Comment by raacampbell** [comment]
Actually, the file you made is better. But I will still make it possible to have different offset types.

File: code/+stitchit/+tileload/illuminationCorrector.m [hunk 1]
```diff
@@ -64,10 +64,8 @@
     % It's a bit of a hack to deal with https://github.com/SainsburyWellcomeCentre/StitchIt/issues/145
     correctIllumOffset=true;
     if correctIllumOffset
```
Comment (raacampbell) [inline code/+stitchit/+tileload/illuminationCorrector.m:66]: I will change this line so it respects userConfig.tile.doIlluminationCorrection

**Comment by raacampbell** [comment]
I am working on this but it might be slow because with the one test acquisition I have, it seems that I get more tile artifacts with illumination correction done your way.

**Comment by raacampbell** [comment]
I think I have something that works now. I'll merge into `dev` soon and ask you to test it.

**Comment by raacampbell** [comment]
OK... I took your changes and used them as the basis of this: d5d556110e2a9d7d02add87a65a907626165a711
Can you try and see if it works for you?
If so, I will tidy it and consider the problem sorted for now. 
In particular I want a way of automatically updating the INI files as it's annoying to have people need to edit them following changes.

**Comment by raacampbell** [comment]
I'm actually a bit unclear where we are here. I got an e-mail saying:

```
Add:
offsetValue = offset.(offsetType);
For now the first stitchAllChannels crashes in tileLoad since there is no offset returned when creating the offset file.
```

I don't see that in this comment thread so I guess it relates to a commit but I don't see where that is.

**Comment by ablot** [comment]
This was a comment on d5d5561 
Sure. I'll close this PR and see if anything needs to be changed in `dev`

**Comment by raacampbell** [comment]
Sure just send me a PR if you need other stuff done. Otherwise let me know if all is good. I still need to add INI file settings but now these populate automatically to the user files from the default file (write permissions allowing).

**Pull request closed without merging**